rapid7/metasploit-framework
# Add in Module for CVE-2021-42237 - Sitecore Experience Platform Pre-Auth Deserialization RCE

Add in an module for CVE-2021-42237. Based heavily off of the original article from https://blog.assetnote.io/2021/11/02/sitecore-rce/, props to AssetNote for finding the bug!

## Verification

1. Start `msfconsole`
1. Do: `use exploit/windows/http/cve_2021_42237`
1. Set the `RHOSTS` option.
1. Set the `TARGET` option if desired.
1. Set the `VHOST` to the virtual host name of the site, which will by default be `NewInstance1`.
1. Set an appropriate payload for the `PAYLOAD` option.
1. Set any additional options as required by the previously selected payload
1. Run the exploit
1. Verify you get a shell as `NT AUTHORITY\NETWORK SERVICE`
1. Optional: Verify you can get a `SYSTEM` shell by running `getsystem -t 4`

Contact me if you need additional help setting up SiteCore; instructions are in the documentation but a few steps are a little tricky.
